snapshot/lib/snapshot/test_command_generator.rb

```diff
@@ -78,9 +78,17 @@ def find_device(device_name, os_version = Snapshot.config[:ios_version])
         end
       end
 
-      def device_udid(device_name, os_version = Snapshot.config[:ios_version])
+      def device_udid(device_name)
+        os = device_name =~ /^Apple TV/ ? "tvOS" : "iOS"
+        os_version = Snapshot.config[:ios_version] || Snapshot::LatestOsVersion.version(os)
+
         device = find_device(device_name, os_version)
 
+        if device.nil?
+          UI.user_error!("No device found named '#{device_name}' for version '#{os_version}'")
```

Comment:
Hey @KrauseFx. It did not report anything before which had cost me a bit of time, thus I think it should definitely tell the user about it.

I'm not familiar with the options for reporting errors/warnings in fastlane so I picked what sounded right and was happy with the output (see below).

Test setup was: 
[wrong_iosversion.zip](https://github.com/fastlane/fastlane/files/479533/wrong_iosversion.zip)
which sets the ios_version to 11.0.

```
 wolfgang  MacBook-Pro-5  ~  Development  stuff  snapshottest  %  ~/Development/stuff/fastlane/snapshot/bin/snapshot
[10:28:03]: $ xcodebuild -list -project ./Test.xcodeproj
2016-09-19 10:28:04.103 xcodebuild[42361:3266287] [MT] PluginLoading: Required plug-in compatibility UUID ACA8656B-FEA8-4B6D-8E4A-93F4C95C362C for plug-in at path '~/Library/Application Support/Developer/Shared/Xcode/Plug-ins/XcodeColors.xcplugin' not present in DVTPlugInCompatibilityUUIDs
2016-09-19 10:28:04.104 xcodebuild[42361:3266287] [MT] PluginLoading: Required plug-in compatibility UUID ACA8656B-FEA8-4B6D-8E4A-93F4C95C362C for plug-in at path '~/Library/Application Support/Developer/Shared/Xcode/Plug-ins/QuickLocalization.xcplugin' not present in DVTPlugInCompatibilityUUIDs
2016-09-19 10:28:04.104 xcodebuild[42361:3266287] [MT] PluginLoading: Required plug-in compatibility UUID ACA8656B-FEA8-4B6D-8E4A-93F4C95C362C for plug-in at path '~/Library/Application Support/Developer/Shared/Xcode/Plug-ins/KPRunEverywhereXcodePlugin.xcplugin' not present in DVTPlugInCompatibilityUUIDs
2016-09-19 10:28:04.105 xcodebuild[42361:3266287] [MT] PluginLoading: Required plug-in compatibility UUID ACA8656B-FEA8-4B6D-8E4A-93F4C95C362C for plug-in at path '~/Library/Application Support/Developer/Shared/Xcode/Plug-ins/Alcatraz.xcplugin' not present in DVTPlugInCompatibilityUUIDs

+----------------------------+------------------------------------------------------------+
|                               Summary for snapshot 1.14.0                               |
+----------------------------+------------------------------------------------------------+
| ios_version                | 11.0                                                       |
| devices                    | ["iPhone 6"]                                               |
| languages                  | ["en-US", "de-DE"]                                         |
| project                    | ./Test.xcodeproj                                           |
| scheme                     | Test                                                       |
| output_directory           | /Users/wolfgang/Development/stuff/snapshottest/screenshots |
| launch_arguments           | [""]                                                       |
| skip_open_summary          | false                                                      |
| clear_previous_screenshots | false                                                      |
| reinstall_app              | false                                                      |
| erase_simulator            | false                                                      |
| localize_simulator         | false                                                      |
| buildlog_path              | ~/Library/Logs/snapshot                                    |
| clean                      | false                                                      |
| number_of_retries          | 1                                                          |
| stop_after_first_error     | false                                                      |
+----------------------------+------------------------------------------------------------+

[10:28:04]: Building and running project - this might take some time...
[10:28:04]: snapshot run 1 of 2
[10:28:04]: Patching '/Users/wolfgang/Library/Preferences/com.apple.iphonesimulator.plist' to scale simulator to 100%
[10:28:05]: No device found named 'iPhone 6' for version '11.0'
[10:28:05]: Tests failed, re-trying 1 out of 2 times
[10:28:05]: Patching '/Users/wolfgang/Library/Preferences/com.apple.iphonesimulator.plist' to scale simulator to 100%
[10:28:06]: No device found named 'iPhone 6' for version '11.0'
[10:28:06]: snapshot run 2 of 2
[10:28:06]: Patching '/Users/wolfgang/Library/Preferences/com.apple.iphonesimulator.plist' to scale simulator to 100%
[10:28:07]: No device found named 'iPhone 6' for version '11.0'
[10:28:07]: Tests failed, re-trying 1 out of 2 times
[10:28:08]: Patching '/Users/wolfgang/Library/Preferences/com.apple.iphonesimulator.plist' to scale simulator to 100%
[10:28:09]: No device found named 'iPhone 6' for version '11.0'

+----------+-------+-------+
|     snapshot results     |
+----------+-------+-------+
| Device   | en-US | de-DE |
+----------+-------+-------+
| iPhone 6 |  ❌    |  ❌    |
+----------+-------+-------+


[!] No device found named 'iPhone 6' for version '11.0'; No device found named 'iPhone 6' for version '11.0'
```